kingpin/actors/group.py

```diff
@@ -328,6 +333,24 @@ class Async(BaseGroupActor):
     finish before the failure is returned.
     """
 
+    all_options = {
+        'concurrency': (int, 0, "Max number of concurrent executions."),
+        'contexts': (list, [], "List of contextual hashes."),
+        'acts': (list, REQUIRED, "Array of actor definitions.")
+    }
+
+    @gen.coroutine
+    def worker(self, act_gen, worker_id):
+        """Syncronous actor-executor that fetches from a shared generator.
+
+        Concurrency specifies the total number of workers. Each worker
+        continuously grabs the next task (actor) and executes it. Since the
+        acts come from the same pool (generator) no actions are executed twice.
+        """
+        for act in act_gen:
+            self.log.debug('Worker[%s]: %s' % (worker_id, act._desc))
+            yield act.execute()
```

Comment:
Todo: figure out a fix for exception handling, such that the end-result is same as `yield [task, task, task]`